debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -635,6 +639,10 @@ protected void createSchemaChangeEventsForTables(ChangeEventSourceContext source
             LOGGER.debug("Processing schema event {}", event);
 
             final TableId tableId = event.getTables().isEmpty() ? null : event.getTables().iterator().next().id();
+            if (snapshottingTask.isBlocking() && !snapshotContext.capturedTables.contains(tableId)) {
```

Comment:
@jpechane Here I need to avoid to resend already processed events and process the only ones requested in bloking snapshot signal. Is there a way to check the same for the `DROP` statements for which `tableId` is null? 

I mean something more robust then checking the name of the table in the `ddl` field.

Reply:
@mfvitale Under which conditions the `DROP` even will not contain `tableId`?

Reply:
I think in any DROP events 

![image](https://github.com/debezium/debezium/assets/2572323/a2cca74c-28fa-4fda-b7c6-9a9c0852e30f)

Reply:
@jpechane Seems the missing of the `tableChanges` depends on this code

![image](https://github.com/debezium/debezium/assets/2572323/58b74986-06ae-44c1-a358-7139aaaaac72)

where the table is added to changes but removed from `tablesByTableId` and then whit will cause 

![image](https://github.com/debezium/debezium/assets/2572323/98cb849a-4f01-4d6e-b72c-5f469732e239)

`tables().forTable(tableId) to return `null`.

Should `tablesByTableId` be aligned with `changes`?

Reply:
@jpechane I think I have found an acceptable solution. Please check commit f478c7e1af0597f0ed7bf6902aabf791e7cdd9e5

Reply:
@mfvitale It is good workaround. But I'd argu that the missing table change is bug in this case. Would you be able to dig into it a bit deeper? If not we can merge this PR and create another Jira just for that problem.

Reply:
I have the same feeling but not 100%. I think in this case is better to create a separate Jira issue to investigate it.